politeiawww/invoices_test.go

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) 2017-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/decred/dcrtime/merkle"
+	"github.com/decred/politeia/politeiad/api/v1/identity"
+	"github.com/decred/politeia/politeiad/api/v1/mime"
+	cms "github.com/decred/politeia/politeiawww/api/cms/v1"
+	www "github.com/decred/politeia/politeiawww/api/www/v1"
+	"github.com/decred/politeia/politeiawww/user"
+	"github.com/decred/politeia/util"
+)
+
+func createValidLineItems(t *testing.T) []cms.LineItemsInput {
+	t.Helper()
+
+	// valid line item entries
+	lineItemLabor := cms.LineItemsInput{
+		Type:          cms.LineItemTypeLabor,
+		Domain:        "Development",
+		Subdomain:     "politeia",
+		Description:   "PR #1",
+		ProposalToken: "",
+		Labor:         40,
+		Expenses:      0,
+	}
+	lineItemExpense := cms.LineItemsInput{
+		Type:          cms.LineItemTypeExpense,
+		Domain:        "Design",
+		Subdomain:     "pgui",
+		Description:   "Artwork",
+		ProposalToken: "",
+		Labor:         0,
+		Expenses:      1000,
+	}
+	lineItemMisc := cms.LineItemsInput{
+		Type:          cms.LineItemTypeMisc,
+		Domain:        "Research",
+		Subdomain:     "dcrd",
+		Description:   "reorg",
+		ProposalToken: "",
+		Labor:         0,
+		Expenses:      10000,
+	}
+
+	return []cms.LineItemsInput{
+		lineItemLabor,
+		lineItemExpense,
+		lineItemMisc,
+	}
+}
+
+func createInvoiceInput(t *testing.T, li []cms.LineItemsInput) cms.InvoiceInput {
+	t.Helper()
+
+	var (
```

Comment:
Why are these being defined here as separate variables instead of just defining them directly in the `InvoiceInput`?